src/platform/packages/shared/shared-ux/chrome/navigation/src/ui/components/navigation_section_ui.tsx

```diff
@@ -135,14 +135,6 @@ const serializeNavNode = (
   return serialized;
 };
 
-const isEuiCollapsibleNavItemProps = (
-  props: EuiCollapsibleNavItemProps | EuiCollapsibleNavSubItemProps
-): props is EuiCollapsibleNavItemProps => {
-  return (
-    props.title !== undefined && (props as EuiCollapsibleNavSubItemProps).renderItem === undefined
-  );
-};
-
```

Comment:
Nav item should work without title or renderItem.

It may be worthwhile to enforce those types of nav items should have `children`, because we only want to use that pattern for "untitled" child nav items: [the items in the footer section](https://github.com/elastic/kibana/pull/218050#issue-2991944557) are an example.